File: 2673. Make Costs of Paths Equal in a Binary Tree/Solution.py
class Solution:
    def minIncrements(self, n: int, cost: List[int]) -> int:
        self.res = 0
        self.dfs(1, cost)
        return self.res
        
    def dfs(self, curr, cost):
        if curr * 2 > len(cost): return 0
        left, right = cost[curr * 2 - 1] + self.dfs(curr * 2, cost), cost[curr * 2] + self.dfs(curr * 2 + 1, cost)
        self.res += abs(left - right)
        return max(left, right)

File: 2673. Make Costs of Paths Equal in a Binary Tree/Solution.cpp
class Solution {
public:
    int res = 0;
    int minIncrements(int n, vector<int>& cost) {
        dfs(1, cost);
        return res;
    }

    int dfs(int curr, vector<int>& cost) {
        if (curr * 2 > cost.size()) return 0;
        int left = cost[curr * 2 - 1] + dfs(curr * 2, cost), right = cost[curr * 2] + dfs(curr * 2 + 1, cost);
        res += abs(left - right);
        return max(left, right);
    }
};